src/tls/incoming.rs

```diff
@@ -98,3 +105,101 @@ impl From<TcpListener> for MaybeTlsListener {
         }
     }
 }
+
+pub struct MaybeTlsIncomingStream<S> {
```

Comment:
Why is this called with `Incoming` on it? This should just be a `MaybeTlsStream`? It can store an Option<PeerAddr> to make it more flexible.

Reply:
So I initially rewrote the existing trivial `MaybeTlsStream` to handle the accept handshake case. However, this ended up complicating the sources with a lot (a dozen?) of `#[cfg]` flags due to the `Accepting` state variant never being constructed when TLS sources are compiled out. So, I restored the original `MaybeTlsStream` which handles the both a raw stream and a successfully negotiated TLS stream, and split off this incoming stream, which handles the negotiation and then drops into a stream state. It reduced the number of lines of code, so I figured it was a win.

Reply:
Is this an issue to do with feature flags? if so I see that as a reason to remove them. Yeah, just seems odd that we need two types for this. It should be quite simple :/

Reply:
Even discounting the feature flags issue, splitting this reduced the lines of code a bit. The diff from before I added all the feature flag gates (so it nulls out adding and removing them):

```
 src/tls/incoming.rs | 123 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++------
 src/tls/mod.rs      |  16 ++++++++--
 src/tls/outgoing.rs |   4 +--
 src/tls/stream.rs   | 135 -------------------------------------------------------------------------------------
 4 files changed, 130 insertions(+), 148 deletions(-)
```

Reply:
Yeah, mostly just find it confusing since for me my mental model is tcpstream wrapped by maybetlstream but with the incoming part it gets a bit fuzzy! but as long as it works I am happy.